### PR TITLE
dns: add dns.rcode keyword v4

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Feature #6621

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6621) ticket: https://redmine.openinfosecfoundation.org/issues/6621

Previous PR: https://github.com/OISF/suricata/pull/10126

Describe changes:
- made the suggested changes from previous PR
- SV tests are running successfully
- copied changes from https://github.com/OISF/suricata/pull/10142/commits/1fc18c11b95864a4cd7bd1c81866055460c4377d to the etc/schema.json file to fix the schema errors from the previous PR
- successfully ran the following tests:
- from rust dir: make check
- from rust dir: cargo test
- from rust dir: cargo clippy --all-features
- from rust dir: rustfmt src/dns/*
- from suricata dir: ./scripts/clang-format.sh check-branch
- from suricata dir: ./scripts/clang-format.sh branch
- from suricata dir running ./src/suricata -u showed that one unit test had failed. 
- from suricata dir running ./src/suricata -u -U dns showed 7 tests passed.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1578

### CODE REVIEW

File: etc/schema.json [hunk 1]
```diff
@@ -1156,7 +1156,12 @@
...
                             "description": "DNS opcode as an integer",
                             "type": "integer"
-                        }
+                        },
+			"authorities": {
+			    "description": "Array of authorities in DNS response",
+			    "$comment": "Probably also need answers and additional records",
+			    "type": "array"
+			}
                     },
```
Comment (inashivb) [inline etc/schema.json:1165]: Is this still what we need or is it https://github.com/OISF/suricata/pull/10147/commits/6286be1cbed2ad219a469d5c78840a8ec973465c @jasonish ?

**Review by inashivb — CHANGES_REQUESTED** [review]
Nice work, @hadiqaalamdar !
CI is all green. Congrats! 🎉 
I'll wait for Jason to tell which schema commit needs to go in.
The only change requested from me is that please don't merge cherry-picked commits by someone else with yours.

Rest, all looks great!

File: etc/schema.json (hunk 1 above)
Comment (jasonish) [reply etc/schema.json:1165]: No, we'll now want the commit from https://github.com/OISF/suricata/pull/10147.

**Comment by victorjulien** [comment]
Can you also add an explanation of the keyword in the git message? Things I'm looking for are:
- that it matches the rcode field
- that it's a integer match
- valid ranges of values
- whether it supports prefilter
- which direction it supports

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10203

**Pull request closed without merging**